src/otx/core/model/entity/classification.py

```
@@ -391,3 +402,79 @@ def _customize_outputs(
             scores=pred_scores,
             labels=pred_labels,
         )
+
+
+class OVHlabelClassificationModel(OVModel):
+    """Hierarchical classification model compatible for OpenVINO IR inference.
+
+    It can consume OpenVINO IR model path or model name from Intel OMZ repository
+    and create the OTX classification model compatible for OTX testing pipeline.
+    """
+
+    def __init__(
+        self,
+        *args,
+        num_multiclass_heads: int = 1,
+        num_multilabel_classes: int = 0,
+        **kwargs,
+    ) -> None:
+        self.num_multiclass_heads = num_multiclass_heads
+        self.num_multilabel_classes = num_multilabel_classes
+        super().__init__(*args, **kwargs)
+
+    def set_hlabel_info(self, hierarchical_info: HLabelInfo) -> None:
+        """Set hierarchical information in model head.
+
+        Since OV IR model consist of all required hierarchy information,
+        this method serves as placehloder
+        """
+        return
+
+    def _create_model(self, *args) -> Model:
+        # confidence_threshold is 0.0 to return scores for all multilabel classes
+        model_api_configuration = {"hierarchical": True, "confidence_threshold": 0.0}
+        return super()._create_model(model_api_configuration)
```

```suggestion
class OVHlabelClassificationModel(OVModel[HlabelClsBatchDataEntity, HlabelClsBatchPredEntity, T_OTXTileBatchDataEntity]):
    """Hierarchical classification model compatible for OpenVINO IR inference.

    It can consume OpenVINO IR model path or model name from Intel OMZ repository
    and create the OTX classification model compatible for OTX testing pipeline.
    """

    def __init__(
        self,
        num_classes: int,
        model_name: str,
        model_type: str,
        async_inference: bool = True,
        max_num_requests: int | None = None,
        use_throughput_mode: bool = True,
        num_multiclass_heads: int = 1,
        num_multilabel_classes: int = 0,
    ) -> None:
        self.num_multiclass_heads = num_multiclass_heads
        self.num_multilabel_classes = num_multilabel_classes
        model_api_configuration = {"hierarchical": True, "confidence_threshold": 0.0}
        super().__init__(num_classes, ..., model_api_configuration)

    def set_hlabel_info(self, hierarchical_info: HLabelInfo) -> None:
        """Set hierarchical information in model head.

        Since OV IR model consist of all required hierarchy information,
        this method raises an error if the input information is not equal to the information written in the OV model
        """
        if hierarchical_info != self.hierarchical_info:
              msg = ...
              raise ValueError(msg)
```

---

What is self.hierarchical_info? OVModel doesn't have `hierarchical_info` attribute

---

Is there any interface to get information needed for hierarchical CLS task from MAPI? What I wanted to is leaving `set_hlabel_info` doing nothing, but failing fast as possible if an user tries to assign an invalid `HLabelInfo` into MAPI.

---

Yes, [added it](https://github.com/openvinotoolkit/training_extensions/pull/2815/commits/f0d0e4b3e8615ac6c911edb5c56cfa4559971a31)